test/code/plugins/npmd_config_lib_test.rb

```diff
@@ -226,4 +234,52 @@ def test_contract_03_diag_data
         assert_equal(NPMContract::DATAITEM_ERR_INVALID_FIELDS, _res, "diag data with invalid fields sent but validation did not give correct error")
         assert_equal("MesSage", _prob, "Diag data invalid field was not properly triaged")
     end
+
+    def test_contract_04_epm_health_data
```

Comment:
test cases here and below just validates the contract, and doesnt do anything functional. I would suggest to add functional tests like above which tests the code flow too

Reply:
After this PR, I will raise another one to include different scenario-based tests.